packages/slate-react/src/components/leaf.tsx

```diff
@@ -40,34 +58,38 @@ const Leaf = (props: {
     return () => {
       editorEl.style.minHeight = 'auto'
     }
-  }, [placeholderRef])
+  }, [placeholderRef, leaf])
 
   let children = (
     <String isLast={isLast} leaf={leaf} parent={parent} text={text} />
   )
 
   if (leaf[PLACEHOLDER_SYMBOL]) {
+    const placeholderProps = {
+      ref: placeholderRef,
+      contentEditable: false,
+      style: {
+        position: 'absolute',
+        pointerEvents: 'none',
+        width: '100%',
+        maxWidth: '100%',
+        display: 'block',
+        opacity: '0.333',
+        userSelect: 'none',
+        fontStyle: 'normal',
+        fontWeight: 'normal',
+        textDecoration: 'none',
+      } as React.CSSProperties,
```

Comment:
And then all this placeholder rendering logic can move out of the `<Leaf>` and be consolidated into the top-level `<Editable>`. 

The only props we'd be passing in to the `renderPlaceholder` would be `children` (which is the `placeholder` prop from the editor) and `ref`. The other things here would just be defined inline in our new `<DefaultPlaceholder>` component.

Reply:
I've made these changes except I have kept the `contentEditable: false` in the props passed to `renderPlaceholder` because removing this would break the editor and make the placeholder editable

I've removed the style prop as suggested but made it an export that people can use it for convenience (since without it the placeholder render will break the editor like so:

![image](https://user-images.githubusercontent.com/1188186/114848075-45877700-9dd6-11eb-9d23-6b8a6a14a67b.png)

Also added an example to show how this can be used.